### ISSUE TITLE
Missing ct.hrl dependency

### ISSUE THREAD

**Issue by 0x6e6562** [ferd/cth_readable#22]
When I try to build this as a standalone rebar project (using d9cefd0), I get the following error:

```
$ DEBUG=1 ./rebar3 compile
===> Evaluating config script "rebar.config.script"
===> Load global config file /root/.config/rebar3/rebar.config
===> Expanded command sequence to be run: [{default,app_discovery},
                                                  {default,install_deps},
                                                  {default,lock},
                                                  {default,compile}]
===> Provider: {default,app_discovery}
===> Provider: {default,install_deps}
===> Verifying dependencies...
===> Provider: {default,lock}
===> Provider: {default,compile}
===> run_hooks("/mnt/hgfs/Workspace/cth_readable", pre_hooks, compile) -> no hooks defined

===> Compiling cth_readable
===> run_hooks("/mnt/hgfs/Workspace/cth_readable", pre_hooks, compile) -> no hooks defined

===> run_hooks("/mnt/hgfs/Workspace/cth_readable", pre_hooks, erlc_compile) -> no hooks defined

===> erlopts [debug_info]
===> files to compile ["/mnt/hgfs/Workspace/cth_readable/src/cth_readable_transform.erl",
                              "/mnt/hgfs/Workspace/cth_readable/src/cth_readable_compact_shell.erl",
                              "/mnt/hgfs/Workspace/cth_readable/src/cth_readable_helpers.erl",
                              "/mnt/hgfs/Workspace/cth_readable/src/cthr.erl",
                              "/mnt/hgfs/Workspace/cth_readable/src/cth_readable_nosasl.erl",
                              "/mnt/hgfs/Workspace/cth_readable/src/cth_readable_shell.erl",
                              "/mnt/hgfs/Workspace/cth_readable/src/cth_readable_lager_backend.erl",
                              "/mnt/hgfs/Workspace/cth_readable/src/cth_readable_failonly.erl"]
===> Compiling src/cthr.erl failed
src/cthr.erl:3: can't find include lib "common_test/include/ct.hrl"; Make sure common_test is in your app file's 'applications' list

src/cthr.erl:83: Warning: function format/4 is unused

===> Compilation failed: {error,
                                    [["src/cthr.erl:3: can't find include lib \"common_test/include/ct.hrl\"; Make sure common_test is in your app file's 'applications' list\n"]],
                                    [["src/cthr.erl:83: Warning: function format/4 is unused\n"]]}

```

This is using rebar 3.9.0 on Erlang/OTP 21 Erts 10.2.3 (Centos 7.6.1810).

Is there something I'm doing wrong?

**Comment by ferd** [comment]
If you have installed Erlang from a package manager, check that you also have all the side libraries required of a standard install, including source files; Linux package managers have a tendency to remove stuff like test frameworks (eunit / ct) or blindly remove header files from there, which means that you won't be able to use the install to write your own tests.

**Comment by 0x6e6562** [comment]
Many thanks for the heads up - I'm using a custom Erlang install, which has skipped some unused libraries, and potentially it has also skipped the test frameworks from OTP. I'll check this and report back.

**Comment by 0x6e6562** [comment]
The absence of `common_test` in Erlang package I was using was indeed the issue.

For note, I also tried the RabbitMQ RPM package for Erlang, which appears to be a minimalist Erlang distribution for CentOS - this also has the same issue as our own custom Erlang install.

After I installed the ESL package, `common_test` is available and now this library compiles.

Just for people googling this, I ran into this issue by trying to bootstrap rebar3 (which transtively depends on `cth_readable`). So not having `common_test` on the load path causes the rebar bootstrap to fail.

The ESL package appears to suffice the rebar dependencies - just be aware that the ESL package can bring in a lot of dependencies in you are not careful.
